src/controllers/movies/moviesrecommended.js

```diff
@@ -227,38 +237,30 @@ function loadSuggestionsTab(view, params, tabContent) {
 }
 
 function getTabs() {
-    return [{
+    const tabList = [{
         name: globalize.translate('Movies')
     }, {
         name: globalize.translate('Suggestions')
-    }, {
-        name: globalize.translate('Trailers')
     }, {
         name: globalize.translate('Favorites')
     }, {
         name: globalize.translate('Collections')
     }, {
         name: globalize.translate('Genres')
     }];
+
+    // Only show the 'Trailers' tab if there is a trailer provider plugin
+    const plugins = getPluginsPreLoaded();
+    if (plugins.some((p) => p.toLowerCase().includes('trailer'))) {
```

Comment:
This is not going to work.

The plugins here are [web plugins](https://github.com/jellyfin/jellyfin-web/tree/master/src/plugins) and not server plugins. We should hide the tab by default and only show it if the api returns any trailers. That should be much easier than trying to conditionally add/remove the tab.

Reply:
Thank you for responding; this is why I had asked the best way to go about this more than a month ago. I am not as familiar with the project as someone who works on it all the time, I'm just a community member trying to fix a small issue.

Reply:
Does this mean the expectation is to call the API to get trailers every single pageload? A big part of why I went with the plugin-detection method was that I wanted to avoid that sort of thing. It did not strike me as a performant way of solving the issue.

Reply:
I've implemented this by querying for a single 'Trailer' record.